vllm/model_executor/weight_utils.py

```diff
@@ -277,6 +287,24 @@ def hf_model_weights_iterator(
             with open(param_path, "rb") as f:
                 param = np.load(f)
             yield name, torch.from_numpy(param)
+    elif load_format == "tensorizer":
+        tensorizer_args = load_format.params
+        warnings.warn(
+            "Deserializing HuggingFace models is not optimized for "
+            "loading on vLLM, as tensorizer is forced to load to CPU. "
+            "Consider deserializing a vLLM model instead for faster "
+            "load times. See the examples/tensorize_vllm_model.py example "
+            "script for serializing vLLM models.",
+            category=PerformanceWarning,
+            stacklevel=2)
+        deserializer_args = tensorizer_args.deserializer_params
+        stream_params = tensorizer_args.stream_params
+        stream = open_stream(tensorizer_args.tensorizer_uri, **stream_params)
+        with TensorDeserializer(stream, **deserializer_args,
+                                device="cpu") as state:
+            for name, param in state.items():
+                yield name, param
+        del state
```

Comment:
This can be done in a followup, but it would be good to refactor the weight_utils to use some sort of an interface for different weight loading methods instead of just having multiple branches

Reply:
I agree, and will assume this is out of the scope of this PR.